.docs/contributing-book/src/documentation/pull-requests/commit.md

```diff
@@ -0,0 +1,48 @@
+# Committing your work
+
+A `commit` can be thought of as a snapshot in time which captures the difference between the snapshot that is currently being made and the previous snapshot.
+
+When creating a snapshot there are some points to consider in order to keep a high quality log:
+
+- The quantity of work between commits
+- Grouping work by task / concept
+- The message used to track the changes between commits
+
+## Quantity of Work
+
+The amount of work done per commit is dependent upon the task that is being solved however there is a general rule to follow and that is to avoid the extremes of committing everything at once or committing every minor change such as a typo.
+
+The reason for not committing all of the work at once is twofold:
+
+- When a fault occurs which leads to a loss of work then all of that work is lost
+- If a section of work needs to be reverted then everything must be reverted
+
+Similarly, small commits should be avoided because:
+
+- A lot of commits may be considered as spam and may be difficult to parse
+
+## Categorization
+
+Categorizing commits into issues being resolved allows us to easily scope the amount of work per commit. With appropriate categories the likelihood of too much, or not enough, work being committed is reduced.
+
+An example could be a failing test suite which includes multiple functions that were re-written. In this instance it may be a good idea to fix a test, or a test suite, for one specific function and committing that work before moving onto the next. 
+
+This creates a clear separation within the task of fixing the test suites by fixing one suite in one commit and another in another commit.
+
+## Commit Messages
+
+Once the issue has been resolved it's time to write a message that will distinguish this commit from any other.
+
+The commit message should be a short sentence summarizing the changes:
+
+- `Good commit message:` 
+  - Fixed a conditional check in the deposit function which did not fail under condition XYZ
```

Comment:
```suggestion
  - Fix a conditional check in the deposit function which did not fail under condition XYZ
```

Reply:
I'm pretty sure people do not say "I fix a conditional check" and instead they say "I have fixed a conditional check". The example above omits the first two words.
I also do not subscribe to the idea of using "Fix, Chore, whatever" in the commit message. People are free to do that but that's only useful when resolving bugs that have sprung up and the log is being unwound. Labels and issues exist for this purpose otherwise.

Reply:
Whether or not we use so-called "semantic commits" is another issue.

The change I suggested here is to use the imperative mood in the commit message ("fix" vs "fixed"), as recommended by pretty much all the guides to writing good commit messages (including the 2 linked in this book section.

Reply:
Git itself uses this convention. After a merge or a revert, the corresponding commit messages are "Merge ..." or "Revert ..." , not "Merged ..." or "Reverted ..."

Reply:
I have read most of the article and my take away is to write concise messages.

I 100% disagree with the imperative tone. It's incorrect and I will not follow it. Who exactly are you commanding when you write such a message? The reader, yourself, the commit? It doesn't make sense.
The process is that you do some work and _describe_ the work done. You do not write a command as if that work now needs to be done because you've literally just done it and committed it. It's a bad standard and it's grammatically incorrect.

Reply:
<img width="754" alt="image" src="https://user-images.githubusercontent.com/17053076/192045791-2097b7f4-9e11-413f-881e-227bedf26f15.png">

Reply:
I think the key here is to think in terms of:
"When merged into master, this patch will ... (i.e: Fix xyz)

Reply:
I wouldn't get too bogged down on this. It's good to have guidelines but ultimately git commit messages are essentially immutable (I think you can `amend` them but it's basically a patch commit on top of the old one).

I think "concise" is a good goal, but taking a look at the average commit message in this repo, we have a ways to go before we start being concerned about tense/mood of the comment.

Unlike guidelines relating to code quality, commit messages should just be on a best-effot basis.

Reply:
They actually think that prepending some text in your mind is acceptable. How about I completely omit a message and they read my entire mind to figure out the commit? It's absolutely absurd that they think it's acceptable and just because github follows it doesn't mean it's good. Look at who has the most influence in society as an example. Is it the correct groups such as philosophers, teachers etc. or dumb celebrities that start mindless trends which everyone follows. The argument that "they do it" is subpar at best. The standard is meant to be judged on its own quality and not adoption.

That being said, I don't care about what someone decides to put in the message as long as it is concise and accurate. Although if someone starts adding emojis or something then that will annoy me.

I need to update the section with the points stated above since it looks like enough eyes have generally agreed. I will likely do that on Sunday so I will mark this back as a draft for now and rerequest a review once the changes are made.

Reply:
> as long as it is concise and accurate.

👍